src/dnx/pal.win32.cpp

```diff
@@ -81,19 +83,56 @@ BOOL GetFullPath(LPCTSTR szPath, LPTSTR pszNormalizedPath)
     return TRUE;
 }
 
+namespace
+{
+    std::wstring get_runtime_path(TraceWriter& trace_writer)
+    {
+        wchar_t program_files_path_buffer[MAX_PATH];
+        wchar_t* program_files_paths[] =
+        {
+            L"ProgramFiles(x86)",
+#if !defined(_M_IX86)   // in case of 32-bit processes both ProgramFiles and ProgramFiles(x86) point to ProgramFiles(x86)
+            L"ProgramFiles"
+#endif
+        };
+
+        for (auto i = 0; i < sizeof(program_files_paths)/sizeof(wchar_t*); i++)
+        {
+            if (GetEnvironmentVariable(program_files_paths[i], program_files_path_buffer, MAX_PATH) != 0)
+            {
+                auto runtime_path = dnx::servicing::get_runtime_path(program_files_path_buffer, trace_writer);
+                if (runtime_path.length() > 0)
+                {
+                    return dnx::utils::path_combine(runtime_path, L"bin\\");
+                }
+            }
+        }
+
+        return std::wstring{};
+    }
+}
+
 int CallApplicationMain(const wchar_t* moduleName, const char* functionName, CALL_APPLICATION_MAIN_DATA* data, TraceWriter traceWriter)
 {
-    HMODULE hostModule;
+    HMODULE hostModule = nullptr;
     try
     {
-        hostModule = LoadLibraryEx(moduleName, NULL, LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
+        auto runtime_path = get_runtime_path(traceWriter);
+        if (runtime_path.length() > 0)
+        {
+            traceWriter.Write(std::wstring(L"Redirecting runtime to: ").append(runtime_path), true);
+            SetEnvironmentVariable(_T("DNX_DEFAULT_LIB"), runtime_path.c_str());
```

Comment:
In a few places but most notably in the app domain manager(s). It's originally set (here)[https://github.com/aspnet/dnx/blob/dev/src/dnx/dnx.cpp#L259] and updated for servicing to force loading managed assemblies from the redirected runtime.